imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -0,0 +1,115 @@
+"""Calculates Extended Raw Events for ULTRA L1b."""
+
+import numpy as np
+
+from imap_processing.ultra.l1b.l1b_extended_config import (
+    D_SLIT_FOIL,
+    N_ELEMENTS,
+    SLIT_Z,
+    TRIG_CONSTANT,
+    YF_ESTIMATE_LEFT,
+    YF_ESTIMATE_RIGHT,
+)
+from imap_processing.ultra.l1b.lookup_utils import (
+    get_image_params,
+    get_y_adjust,
+)
+
+# TODO: make lookup tables into config files.
+
+
+def get_front_x_position(
+    start_type: np.array, start_position_tdc: np.array
+) -> np.array:
+    """
+    Calculate the front xf position.
+
+    Converts Start Position Time to Digital Converter (TDC)
+    values into units of hundredths of a millimeter using a scale factor and offsets.
+    Further description is available on pages 30 of
+    IMAP-Ultra Flight Software Specification document (7523-9009_Rev_-.pdf).
+
+    Parameters
+    ----------
+    start_type : np.array
+        Start Type: 1=Left, 2=Right.
+    start_position_tdc : np.array
+        Start Position Time to Digital Converter (TDC).
+
+    Returns
+    -------
+    xf : np.array
+        X front position (hundredths of a millimeter).
+    """
+    # Left and right start types.
+    indices = np.where((start_type == 1) | (start_type == 2))
```

Comment:
numpy documentation suggests the use of `np.nonzero()` when used to get indices like here. The below use of `where` with all three arguments supplied is my take on what the documentation says is the intended use.